torch/csrc/jit/passes/inline_autodiff_subgraphs.cpp

```diff
@@ -14,7 +14,8 @@ namespace jit {
 bool canRunWithAutograd(Node* node) {
   auto kind = node->kind();
   return kind != prim::FusionGroup && kind != prim::CudaFusionGroup &&
-      kind != prim::TensorExprGroup && (kind.is_aten() || kind.is_prim());
+      kind != prim::TypeCheck && kind != prim::TensorExprGroup &&
```

Comment:
Can we change https://github.com/pytorch/pytorch/blob/3fed596cb367e2a8dde858acd5ed5899303de430/torch/csrc/jit/passes/inline_autodiff_subgraphs.cpp#L44 to recurse through the nodes? It's misleading to say that prim::TypeCheck can't be run with Autograd

Reply:
I added the recursion but also kept `prim::TypeCheck` per your comment below.